### PR TITLE
refactor(web): asset interaction

### PR DESCRIPTION
**Pull request by michelheusschen**
- replace store with runes
- centralize derived logic like `isAllTrashed`, `isAllUserOwned`, etc.
- add some tests

### CODE REVIEW

File: web/src/lib/components/photos-page/asset-grid.svelte [hunk 1]
```diff
@@ -459,15 +457,15 @@
...
+    const ids = await stackAssets(assetInteraction.selectedAssetsArray);
     if (ids) {
       $assetStore.removeAssets(ids);
       onEscape();
     }
   };
 
   const toggleArchive = async () => {
-    const ids = await archiveAssets(Array.from($selectedAssets), !isAllArchived);
+    const ids = await archiveAssets(assetInteraction.selectedAssetsArray, assetInteraction.isAllArchived);
```
Comment (danieldietzler) [inline web/src/lib/components/photos-page/asset-grid.svelte:468]: You missed the *not* here, right? `!assetInteraction.isAllArchived`

File: web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte [hunk 2]
```diff
@@ -32,33 +32,26 @@
...
-  let isLinkActionAvailable = $state(false);
-
-  $effect(() => {
-    const selection = [...$selectedAssets];
-    isAllOwned = selection.every((asset) => asset.ownerId === $user.id);
-    isAllFavorite = selection.every((asset) => asset.isFavorite);
-    isAssetStackSelected = selection.length === 1 && !!selection[0].stack;
-    const isLivePhoto = selection.length === 1 && !!selection[0].livePhotoVideoId;
+  let selectedAssets = $derived(assetInteraction.selectedAssetsArray);
+  let isAllOwned = $derived(selectedAssets.every((asset) => asset.ownerId === $user.id));
```
Comment (danieldietzler) [inline web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte:38]: Can you replace this as well?

File: web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte [hunk 3]
```diff
@@ -32,33 +32,26 @@
...
+    const isLivePhoto = selectedAssets.length === 1 && !!selectedAssets[0].livePhotoVideoId;
     const isLivePhotoCandidate =
-      selection.length === 2 &&
-      selection.some((asset) => asset.type === AssetTypeEnum.Image) &&
-      selection.some((asset) => asset.type === AssetTypeEnum.Image);
-    isLinkActionAvailable = isAllOwned && (isLivePhoto || isLivePhotoCandidate);
-  });
+      selectedAssets.length === 2 &&
+      selectedAssets.some((asset) => asset.type === AssetTypeEnum.Image) &&
+      selectedAssets.some((asset) => asset.type === AssetTypeEnum.Image);
```
Comment (danieldietzler) [inline web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte:45]: I realize that this has been duplicated before as well. But why? Am I overlooking something or are those the exact same conditions?

**Review by danieldietzler — CHANGES_REQUESTED** [review]

File: web/src/lib/components/photos-page/asset-grid.svelte (hunk 1 above)
Comment (michelheusschen) [reply web/src/lib/components/photos-page/asset-grid.svelte:468]: Yup, well spotted

File: web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte (hunk 3 above)
Comment (michelheusschen) [reply web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte:44]: This is indeed wrong, I'll open a PR to fix it

File: web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte [hunk 4]
```diff
@@ -32,33 +32,25 @@
...
-  let isAssetStackSelected = $state(false);
-  let isLinkActionAvailable = $state(false);
-
-  $effect(() => {
-    const selection = [...$selectedAssets];
-    isAllOwned = selection.every((asset) => asset.ownerId === $user.id);
-    isAllFavorite = selection.every((asset) => asset.isFavorite);
-    isAssetStackSelected = selection.length === 1 && !!selection[0].stack;
-    const isLivePhoto = selection.length === 1 && !!selection[0].livePhotoVideoId;
+  let selectedAssets = $derived(assetInteraction.selectedAssetsArray);
```
Comment (danieldietzler) [inline web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte:37]: This variable is never written to or used. So, pretty sure we can just remove it completely

**Review by danieldietzler — APPROVED** [review]
Code looks good to me and I clicked around in the UI and tried various sorts of selections/asset interactions without any issues.

File: web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte (hunk 4 above)
Comment (michelheusschen) [reply web/src/routes/(user)/photos/[[assetId=id]]/+page.svelte:37]: It's not strictly necessary, but it improves the readability of the deriveds below that use this variable, so I’d prefer to keep it.

**Review by jrasm91 — APPROVED** [review]

**Pull request merged**